src/actions/roles.test.js

```diff
@@ -30,36 +29,28 @@ describe('Actions.Roles', () => {
         nock(Client4.getRolesRoute()).
             post('/names').
             reply(200, [TestHelper.basicRoles.system_admin]);
+
         await Actions.getRolesByNames(['system_admin'])(store.dispatch, store.getState);
 
         const state = store.getState();
-        const request = state.requests.roles.getRolesByNames;
         const {roles} = state.entities.roles;
 
-        if (request.status === RequestStatus.FAILURE) {
-            throw new Error(JSON.stringify(request.error));
-        }
-
-        assert.equal(roles.system_admin.name, 'system_admin');
-        assert.deepEqual(roles.system_admin.permissions, TestHelper.basicRoles.system_admin.permissions);
+        assert.strictEqual(roles.system_admin.name, 'system_admin');
```

Comment:
@reflog When I try running these tests, I get an error saying `TypeError: Cannot read property 'name' of undefined`. It looks to me that `const {roles} = state.entities.roles;` is returning an empty object. Do you have any ideas on how to fix this?

Reply:
@djanda97 this happens due to a bug in your removal of 
```
        onSuccess: [RoleTypes.RECEIVED_ROLES, RoleTypes.ROLES_BY_NAMES_SUCCESS],
```
and other onSuccess parts.
Previously, it would issue two actions RoleTypes.RECEIVED_ROLES, RoleTypes.ROLES_BY_NAMES_SUCCESS on success, they are caught in the reducer and roles are populated.
Since you removed the 'request' tracking bit - you need to issue those actions manually if request was successful.